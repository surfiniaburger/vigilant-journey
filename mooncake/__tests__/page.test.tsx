import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import Page from '@/app/page'
 
describe('Page', () => {
  it('renders the main content', () => {
    render(<Page />)
 
    const button = screen.getByRole('button')
 
    expect(button).toBeInTheDocument()
  })

  it('renders homepage unchanged', () => {
    const { container } = render(<Page />)
    expect(container).toMatchSnapshot()
  })
})
