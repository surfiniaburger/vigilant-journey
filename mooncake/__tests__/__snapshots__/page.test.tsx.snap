// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Page renders homepage unchanged 1`] = `
<div>
  <main
    class="flex h-screen items-center justify-center"
  >
    <div
      class="w-full max-w-lg h-[70vh] border rounded-lg shadow-lg"
    >
      <div
        class="w-full h-full flex flex-col"
      >
        <div
          class="relative overflow-y-auto flex-1"
          role="log"
        >
          <div
            style="height: 100%; width: 100%;"
          >
            <div
              class="p-4"
            />
          </div>
        </div>
        <div
          class="p-4 bg-background border-t"
        >
          <button
            aria-label="Voice Button"
            class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive border bg-background shadow-xs hover:bg-accent hover:text-accent-foreground dark:bg-input/30 dark:border-input dark:hover:bg-input/50 size-9 gap-2 transition-all duration-200 relative"
            data-slot="button"
            type="button"
          >
            <div
              class="flex shrink-0 items-center justify-center overflow-hidden transition-all duration-300 absolute inset-0 rounded-sm bg-muted/50 border-0"
            />
          </button>
        </div>
      </div>
    </div>
  </main>
</div>
`;
